pkgs/applications/office/qownnotes/default.nix

```diff
@@ -1,29 +1,69 @@
-{ mkDerivation, lib, stdenv, fetchurl
-, qmake, qttools, qtbase, qtsvg, qtdeclarative, qtxmlpatterns, qtwebsockets
-, qtx11extras, qtwayland
+{ lib
+, stdenv
+, fetchurl
+, qmake
+, qttools
+, qtbase
+, qtdeclarative
+, qtsvg
+, qtwayland
+, qtwebsockets
+, qtx11extras
+, qtxmlpatterns
+, makeWrapper
+, wrapQtAppsHook
 }:
 
-mkDerivation rec {
+let
   pname = "qownnotes";
+  appname = "QOwnNotes";
   version = "22.12.3";
+in
+stdenv.mkDerivation {
+  inherit pname appname version;
 
   src = fetchurl {
     url = "https://download.tuxfamily.org/${pname}/src/${pname}-${version}.tar.xz";
-    # Fetch the checksum of current version with curl:
-    # curl https://download.tuxfamily.org/qownnotes/src/qownnotes-<version>.tar.xz.sha256
     sha256 = "sha256-fpI7RYOGmWwmau6tF8FPmY2/FtN9foWRX8/WgrNU6E8=";
   };
 
-  nativeBuildInputs = [ qmake qttools ];
+  dontWrapQtApps = true;
 
-  buildInputs = [ qtbase qtsvg qtdeclarative qtxmlpatterns qtwebsockets qtx11extras ]
-    ++ lib.optionals stdenv.isLinux [ qtwayland ];
+  preFixup = ''
+      wrapQtApp "$out/bin/${appname}"
+  '';
+
+  nativeBuildInputs = [
+    qmake
+    qttools
+  ] ++ lib.optionals stdenv.isDarwin [ makeWrapper ]
+  ++ lib.optionals stdenv.isLinux [ wrapQtAppsHook ];
+
+  buildInputs = [
+    qtbase
+    qtdeclarative
+    qtsvg
+    qtwayland
+    qtwebsockets
+    qtx11extras
+    qtxmlpatterns
+  ];
+
+  postInstall = ''
+    ln -s $out/bin/${appname} $out/bin/${pname}
```

Comment:
Maybe we can just set `pname = QOwnNotes`? Not sure though.

Reply:
I don't think so. I just added the symlink since I think it is just beneficial to have a lowercase binary in the path as well.
What do you think? A simple symlink shouldn't hurt, right?

Reply:
Then please symlink the darwin binary (to `$out/bin/${appname}`) before `ln -s $out/bin/${appname} $out/bin/${pname}`, so that we have two files in `$out/bin` under both Linux and macOS.

Reply:
We can't have that. Since macOS uses a case-insensitive file system you cannot have `$out/bin/QOwnNotes` and `$out/bin/qownnotes`.

Linux users are typically used to lowercase binaries, so IMO it makes absolute sense to have a symlink, as I've been using QOwnNotes for quite some time and have often needed my ZSH autocomplete to tell me, that `qownnotes` should be `QOwnNotes` when trying to launch it from the terminal.

The proposed solution means that we have `$out/bin/QOwnNotes` and `$out/bin/qownnotes` on Linux and  the `$out/bin/qownnotes` wrapper on macOS to call it from the terminal.

The application itself would still be QOwnNotes on macOS, so in Spotlight, Alfred etc. it would show up like that.